psyneulink/core/scheduling/condition.py

```diff
@@ -152,3 +155,129 @@ def _dict_summary(self):
         )
     ]
 }
+
+
+class Threshold(graph_scheduler.condition._DependencyValidation, Condition):
+    """Threshold
+
+    Attributes:
+
+        dependency
+            the node on which the Condition depends
+
+        parameter
+            the name of the parameter of **dependency** whose value is
+            to be compared to **threshold**
+
+        threshold
+            the fixed value compared to the value of the **parameter**
+
+        comparator
+            the string comparison operator determining the direction or
+            type of comparison of the value of the **parameter**
+            relative to **threshold**
+
+        indices
+            if specified, a series of indices that reach the desired
+            number given an iterable value for **parameter**
+
+        atol
+            absolute tolerance for the comparison
+
+        rtol
+            relative tolerance (to **threshold**) for the comparison
+
+
+    Satisfied when:
+
+        The comparison between the value of the **parameter** and
+        **threshold** using **comparator** is true
+
+    Notes:
+
+        The comparison must be done with single numbers. If the value of
+        **parameter** contains more than one item, **indices** must be
+        specified.
+    """
+
+    def __init__(
+        self,
+        dependency,
+        parameter,
+        threshold,
+        comparator,
+        indices=None,
+        atol=0,
+        rtol=0,
+    ):
+        if comparator not in comparison_operators:
+            raise graph_scheduler.ConditionError(
+                f'Operator must be one of {list(comparison_operators.keys())}'
+            )
+
+        if parameter not in dependency.parameters:
+            raise graph_scheduler.ConditionError(
+                f'{dependency} has no {parameter} parameter'
+            )
+
+        if (
+            indices is not None
+            and not isinstance(indices, graph_scheduler.TimeScale)
+            and not isinstance(indices, collections.abc.Iterable)
+        ):
+            indices = [indices]
```

Comment:
You'd only need a TimeScale as index for num_executions, and in the python version, that's a Time object which doesn't get a timescale value by int, but in the compiled version, num_executions is an arraytype which needs the list of ints

Reply:
You're already checking `isinstance(indices, graph_scheduler.TimeScale)`,
so you can just have:
```
if (isinstance(indices, TimeScale):
   indices = [indices.val]
```

similarly for iterable. The point is that since you're already doing these checks here the validation can be more complete so you can avoid some of the same checks in condition evaluation.
something like this:
```
if indices is None:
   indices = []
elif (isinstance(indices, TimeScale):
   indices = [indices.val]
elif (isinstance(indices, Iterable):
   indices = [int(i) for i in indices]
```
can the simplify `func`:
```
param_value = getattr(....)
# indices is now guaranteed to be a list of ints
for i in indices:
    param_value = param_value[i]
```

Reply:
Ok so if I'm getting this right, the reason it can't currently be preprocessed like this is that `Time` doesn't implement __getitem__, so both `Time()[TimeScale.RUN]` and `Time()[0]` would throw an exception. But it wouldn't be hard to add this

Reply:
ah, so the list of ints was only required from compilation?
now I get why it needs to be in two places, sorry, missed that the first time.

Reply:
yes, exactly